### PR TITLE
Add M3QL changed stage implementation

### PR DESCRIPTION
**Pull request by ZiwenWan**
## Summary
This PR implements the M3QL `changed` stage that outputs 1 when a value changes from the previous non-null value and 0 when the value is null or unchanged.

## Changes
- **ChangedStage**: Core processing logic that tracks previous non-null values
- **ChangedPlanNode**: M3QL query planning support  
- **Constants.java**: Registered 'changed' function and removed from unimplemented list
- **Factories & Visitors**: Updated PipelineStageFactory, M3PlanNodeFactory, M3PlanVisitor, and SourceBuilderVisitor
- **Comprehensive Tests**: Unit tests for both stage and plan node with edge case coverage

## Functionality
The `changed` stage processes each sample in a time series:
- **First sample**: Always outputs 0 (no previous value to compare)
- **Null/NaN values**: Always output 0
- **Valid values**: Output 1 if different from last non-null value, 0 if same
- **Does not support concurrent segment search** (requires prior sample information)

## Usage
```
fetch metrics | changed
```

## Test Coverage
- Basic functionality with changing and non-changing values
- NaN/null value handling
- Edge cases (single sample, empty series, all NaN values)
- Serialization/deserialization testing
- Plan node creation and visitor pattern

## Test Results
- ✅ Full test suite passes
- ✅ Precommit checks pass
- ✅ Code formatting applied
- ✅ All quality gates passed

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/test/java/org/opensearch/tsdb/lang/m3/stage/ChangedStageTests.java [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+            new FloatSample(3000L, 20.0),  // changed from 10 -> 1 (2000L is missing, treated as null)
+            new FloatSample(5000L, 20.0)   // same as previous -> 0 (4000L is missing, treated as null)
+        );
+        TimeSeries series = new TimeSeries(samples, labels, 1000L, 5000L, 1000L, null);
+        List<TimeSeries> result = stage.process(List.of(series));
+
+        assertEquals(1, result.size());
+
+        // Output should have samples for ALL timestamps in the range [1000L, 5000L] with step 1000L
+        List<Sample> expectedSamples = List.of(
```
Comment (Jinny-Wang) [inline src/test/java/org/opensearch/tsdb/lang/m3/stage/ChangedStageTests.java]: What would be a use case for this? Where input time series does not have any samples but we wanna output all 0s at each timestamp

Comment (ZiwenWan) [reply src/test/java/org/opensearch/tsdb/lang/m3/stage/ChangedStageTests.java]: My understanding is not emitting metrics is a state, and if there is always no metrics emitted, it means the system state is unchanged during the query range. For example, if no error metrics emitted, it means the system remains healthy during this time.

File: src/main/java/org/opensearch/tsdb/lang/m3/stage/ChangedStage.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public boolean supportConcurrentSegmentSearch() {
+        // This stage requires information from prior samples, so it cannot support
+        // concurrent segment search
+        return false;
+    }
```
Comment (venketep-prasad) [inline src/main/java/org/opensearch/tsdb/lang/m3/stage/ChangedStage.java]: Isn't this by default false?

File: src/test/java/org/opensearch/tsdb/lang/m3/stage/ChangedStageTests.java [hunk 3]
```diff
@@ -0,0 +1,299 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.tsdb.lang.m3.stage;
```
Comment (venketep-prasad) [inline src/test/java/org/opensearch/tsdb/lang/m3/stage/ChangedStageTests.java]: Also, include a test in the IT with the `changed` function to test the sourcebuildervisitor changes as well?

**Review by philiplhchan — APPROVED** [review]
